mod/item.php

```diff
@@ -175,6 +175,19 @@ function item_post(&$a) {
 	$app         = ((x($_REQUEST,'source'))      ? strip_tags($_REQUEST['source'])  : '');
 	$extid       = ((x($_REQUEST,'extid'))       ? strip_tags($_REQUEST['extid'])  : '');
 
+	// Check for multiple posts with the same message id (when the post was created via API)
+	if (($message_id != "") AND ($profile_uid != 0)) {
```

Comment:
Concerning " or ' the situation is complicated, I think. There are use cases where " is needed (when building an SQL query, when using control characters like \r or \n, ...

Of course we could agree that we should use ' in every place where these situations doesn't occur, like in array field names.